src/cmap/status.rs

```diff
@@ -62,10 +56,11 @@ impl PoolGenerationSubscriber {
         timeout: std::time::Duration,
     ) -> Option<u32> {
         crate::RUNTIME
-            .timeout(timeout, self.receiver.recv())
+            .timeout(timeout, self.receiver.changed())
```

Comment:
The new API for the `watch` channel no longer yields a value on the receiving end directly. Instead, calling `changed` will await until a new value has been sent since the last time `changed` was called, and then `borrow` can be used to access the current value.